src/workerd/util/sqlite.h

```diff
@@ -522,11 +544,64 @@ class SqliteDatabase::Query final: private ResetListener {
     }
   }
 
+  struct QueryEvent {
+    explicit QueryEvent(Query& q)
+        : parent(q),
+          observer(q.db.sqliteObserver),
+          dbWalSizeBefore(observer.getDbWalSize()),
+          startTime(observer.now()) {}
+
+    ~QueryEvent() {
+      uint64_t dbWalSizeAfter = observer.getDbWalSize();
+      uint64_t dbWalBytesWritten = (dbWalSizeAfter - dbWalSizeBefore);
+      kj::Duration queryLatency = observer.now() - startTime;
+
+      observer.reportQueryEvent(/*queryStatement=*/queryStatement,
+          /*queryRowsRead=*/rowsRead,
```

Comment:
Not sure the comments give us much info here. The variable names are already descriptive enough.